### ISSUE TITLE
Code and pre trained model release

### ISSUE THREAD

**Issue by shloak2611** [facebookresearch/UnsupervisedQA#1]
When can we expect the code to run and pre-trained to be released? Do you have a tentative date?

**Comment by patrick-s-h-lewis** [comment]
Thanks for your interest! I'm hoping to be done with the code in the next two weeks or so

**Comment by askaydevs** [comment]
Great work. Waiting for the implementation details and code.

**Comment by patrick-s-h-lewis** [comment]
Hoping to release today or tomorrow

**Comment by kayoyin** [comment]
Thank you for the great paper! I am interested in the seq2seq model you use for Cloze Translation, where can I find the code to build the model?

**Comment by patrick-s-h-lewis** [comment]
Hey Kayoyin,

Our readme gives a few details about how to go about training your own Translators here:
https://github.com/facebookresearch/UnsupervisedQA#training-your-own-question-translation-models. We used UnsupervisedMT to train our seq2seq cloze translators. We made no significant changes to the codebase, and instead of training the model to tranlslate between a pair of languages, we replaced with the training corpora with cloze questions and natural questions. 

TL;DR:
We used [UnsupervisedMT](https://github.com/facebookresearch/UnsupervisedMT) for our main experiments. But the newer (XLM repo)[https://github.com/facebookresearch/XLM] may be a better place to start - it has more features and is faster.

**Comment by askaydevs** [comment]
`No matching distribution found for torch==0.4.1.post2`

Might it be okay to install 0.4.1 instead?

**Comment by patrick-s-h-lewis** [comment]
should be ok

**Comment by askaydevs** [comment]
`requests.exceptions.ChunkedEncodingError: ("Connection broken: ConnectionResetError(104, 'Connection reset by peer')", ConnectionResetError(104, 'Connection reset by peer'))`

I was generating synthetic QA data with `example_input.txt`. However, my device is not `cuda` enabled. After `82%` completion of `example_input.txt`, this error popped up. I was running the code on a local machine.

Is this a `cuda` issue?

**Comment by askaydevs** [comment]
ubuntu 18.04 is not an officially supported platform for CUDA 9.0, or CUDA 9.2

[https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html](url)

Pl, look into it.

**Comment by patrick-s-h-lewis** [comment]
> requests.exceptions.ChunkedEncodingError: ("Connection broken: ConnectionResetError(104, 'Connection reset by peer')", ConnectionResetError(104, 'Connection reset by peer'))

This is a networking issue ("requests" is a http library for python). My guess is allennlp's server was overloaded, or your internet connection was interrupted, probably not an issue with this codebase, and definitely not a CUDA issue. 

> ubuntu 18.04 is not an officially supported platform for CUDA 9.0, or CUDA 9.2

Good spot, I think it should still install. I am AFK this week so will be slow replying to issues.

Please open new issue tickets for each problem you encounter, closing.

**Comment by askaydevs** [comment]
👍
